common/utils/typeconverter.go

```diff
@@ -13,11 +13,11 @@ func StringSliceToListSegmenterValue(values *[]string) *_segmenters.ListSegmente
 	if values == nil {
 		return nil
 	}
-	segmenterValues := []*_segmenters.SegmenterValue{}
-	for _, val := range *values {
-		segmenterValues = append(segmenterValues, &_segmenters.SegmenterValue{
-			Value: &_segmenters.SegmenterValue_String_{String_: val},
-		})
+	segmenterValues := make([]*_segmenters.SegmenterValue, len(*values))
+	for i := 0; i < len(*values); i++ {
+		segmenterValues[i] = &_segmenters.SegmenterValue{
+			Value: &_segmenters.SegmenterValue_String_{String_: (*values)[i]},
+		}
```

Comment:
This is a series of minor changes to initialise a fixed-length slice instead of a dynamic length slice to store the segmenter values so as to prevent repeated allocation of memory to copy-paste the slice values when `append`-ing.